src/differentials/thunks.jl

```diff
@@ -19,6 +19,74 @@ Base.:(==)(a::AbstractThunk, b::AbstractThunk) = unthunk(a) == unthunk(b)
 Base.:(==)(a::AbstractThunk, b) = unthunk(a) == b
 Base.:(==)(a, b::AbstractThunk) = a == unthunk(b)
 
+Base.:(-)(a::AbstractThunk) = - unthunk(a)
+Base.:(-)(a::AbstractThunk, b) = unthunk(a) - b
+Base.:(-)(a, b::AbstractThunk) = a - unthunk(b)
+Base.:(/)(a::AbstractThunk, b) = unthunk(a) / b
+Base.:(/)(a, b::AbstractThunk) = a / unthunk(b)
```

Comment:
I guess we don't have to worry about both being thunks because that would be a nonlinear operation between two tangent types?

Reply:
Generally I only defined the functions I needed to get the ChainRules tests to pass, and didn't worry too much about all possible combinations. Not sure how less lightweight ChainRulesCore is going to get once we define all these functions? Or is that not a concern at all?

Reply:
> Not sure how less lightweight ChainRulesCore is going to get once we define all these functions? Or is that not a concern at all?

I don't think it is too much of a concern. Not compared to correctness and ease of use.
Except for if the time increase is huge.

E.g. for me current release of ChainRulesCore is
```
julia> @time using ChainRulesCore
  0.251466 seconds (452.55 k allocations: 26.666 MiB)
```
and this branch which adds a ton already gets it up to
```
julia> @time using ChainRulesCore
  0.273390 seconds (486.60 k allocations: 28.348 MiB)
```

I suspect we can get it down more with some precompilation.
If we have to we can move them out to ChainRules.jl (and then require people load ChainRules.jl for testing etc).
But for now lets not worry too much